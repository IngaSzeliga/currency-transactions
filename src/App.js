import React from "react";
import NewTransaction from "./components/NewTransaction";
import "./App.scss";

function App() {
  return (
    <div className="app-container">
      <NewTransaction />
    </div>
  );
}

export default App;
