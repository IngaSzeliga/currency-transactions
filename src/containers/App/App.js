import React, { PureComponent } from "react";
import NewTransaction from "../../components/NewTransaction";
import TransactionsHistory from "../../containers/TransactionsHistory";
import "./App.scss";

class App extends PureComponent {
  state = {
    transactions: []
  };

  addTransaction = transaction => {
    this.setState(previousState => {
      const { transactions } = previousState;
      return {
        transactions: [...transactions, transaction]
      };
    });
  };

  render() {
    const { transactions } = this.state;
    return (
      <div className="app-container">
        <NewTransaction addTransaction={this.addTransaction} />
        <TransactionsHistory transactions={transactions} />
      </div>
    );
  }
}

export default App;
