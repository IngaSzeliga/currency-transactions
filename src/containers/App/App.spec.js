import React from "React";
import { shallow, mount } from "enzyme";
import App from ".";

describe("App component", () => {
  const component = shallow(<App />);

  it("has working component", () => {
    expect(component).toBeDefined();
  });

  it("expects component to never be changed", () => {
    expect(component).toMatchSnapshot();
  });

  it("expects to call deleteTransaction", () => {
    const componentMounted = mount(<App />);
    componentMounted.setState({
      transactions: [
        {
          id: "2caf5711-2913-4e41-95db-088625acf1bf",
          title: "Name",
          amount: 13,
          convertedAmount: 54.99
        }
      ]
    });
    componentMounted
      .instance()
      .deleteTransaction("2caf5711-2913-4e41-95db-088625acf1bf");
    expect(componentMounted.state("transactions")).toEqual([]);
  });

  it("expects to call addTransaction", () => {
    const componentMounted = mount(<App />);
    const transaction = {
      id: "2caf5711-2913-4e41-95db-088625acf1bf",
      title: "Name",
      amount: 13,
      convertedAmount: 54.99
    };
    componentMounted.instance().addTransaction(transaction);
    expect(componentMounted.state("transactions")).toEqual([transaction]);
  });
});
