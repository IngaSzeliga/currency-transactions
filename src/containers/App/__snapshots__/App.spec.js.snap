// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component expects component to never be changed 1`] = `
<div
  className="app-container"
>
  <Memo(Header) />
  <NewTransaction
    addTransaction={[Function]}
  />
  <Memo(TransactionsHistory)
    deleteTransaction={[Function]}
    transactions={Array []}
  />
</div>
`;
